src/hypothesis/core.py

```diff
@@ -812,6 +887,36 @@ def wrapped_test(*arguments, **kwargs):
             )
             arguments, kwargs, test_runner, search_strategy = processed_args
 
+            state = StateForActualGivenExecution(
+                test_runner, search_strategy, test, settings, random)
+
+            reproduce_failure = \
+                wrapped_test._hypothesis_internal_use_reproduce_failure
+
+            if reproduce_failure is not None:
+                try:
+                    state.execute(ConjectureData.for_buffer(
+                        decode_failure(reproduce_failure)),
+                        print_example=True, is_final=True,
+                    )
+                    raise DidNotReproduce(
+                        'Expected the test to raise an error, but it '
+                        'completed successfully.'
+                    )
+                except StopTest:
+                    raise DidNotReproduce(
+                        'The shape of the test has changed in some way '
+                        'from where this blob was defined. Are you sure '
+                        "you're running the same test? This could also be "
+                        'caused by using different versions of Hypothesis.'
+                    )
+                except UnsatisfiedAssumption:
+                    raise DidNotReproduce(
+                        'The test data failed to satisfy an assumption in the '
```

Comment:
I'm tempted to include a hash of `inspect.getsource`, so we can reject known-different tests.

For usablilty, skip the local check if CI can't find the source (write all-zero 'hash'), and if local can't find source assume it's the same.